examples/finnegans-wake-concise-demo.py

```diff
@@ -1,42 +1,37 @@
-# This is an example of Alice setting a Policy on the NuCypher network.
-# In this example, Alice uses n=3.
-
-# WIP w/ hendrix@3.1.0
-
 import datetime
 import os
 import shutil
-import sys
+from os.path import abspath, dirname
 
 import maya
-from twisted.logger import ILogObserver
 from twisted.logger import globalLogPublisher
-from umbral.keys import UmbralPublicKey
-######################
-# Boring setup stuff #
-######################
-from zope.interface import provider
 
+from nucypher.utilities.logging import simpleObserver
 from nucypher.characters.lawful import Alice, Bob, Ursula
-from nucypher.config.constants import SeednodeMetadata
 from nucypher.data_sources import DataSource as Enrico
-# This is already running in another process.
 from nucypher.network.middleware import RestMiddleware
+from umbral.keys import UmbralPublicKey
 
 
-@provider(ILogObserver)
-def simpleObserver(event):
-    print(event)
+######################
+# Boring setup stuff #
+######################
+
+BOOK_PATH = os.path.join(dirname(abspath(__file__)), 'resources', 'finnegans-wake.txt')
 
+# Twisted Logger
 globalLogPublisher.addObserver(simpleObserver)
 
-# Temporary storage area for demo
-SHARED_CRUFTSPACE = "{}/examples-runtime-cruft".format(os.path.dirname(os.path.abspath(__file__)))
-CRUFTSPACE = "{}/finnegans-wake-demo".format(SHARED_CRUFTSPACE)
-CERTIFICATE_DIR = "{}/certs".format(CRUFTSPACE)
-shutil.rmtree(CRUFTSPACE, ignore_errors=True)
-os.mkdir(CRUFTSPACE)
-os.mkdir(CERTIFICATE_DIR)
+# Temporary file storage
+TEMP_FILES_DIR = "{}/examples-runtime-cruft".format(os.path.dirname(os.path.abspath(__file__)))
```

Comment:
Do we want to keep it in `examples`, or do we want to do it in `/tmp`?

Reply:
Good Question.  This is the last type of file storage we have not built an in-memory backend for.  This changed *slightly* in a recent PR, but ultimately we can use something purpose-built here.

Reply:
:+1: for `/tmp`. I'm not sure if we need a custom interface for it there. It could be left open to another PR, should this be an open issue?

Reply:
Yeah, this could be slightly off-topic here, so a new issue perhaps

Reply:
:+1:

Reply:
https://github.com/nucypher/nucypher/issues/525